pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -136,6 +109,6 @@ public static int hashCodeOf(int previousHashCode, int value) {
   }
 
   public static int hashCodeOf(int previousHashCode, long value) {
-    return 31 * previousHashCode + (int) (value ^ (value >>> 32));
+    return 31 * previousHashCode + (int) ((value & 0xFFFFFFFFL) ^ (value >>> 32));
```

Comment:
http://stackoverflow.com/questions/113511/best-implementation-for-hashcode-method